.github/workflows/windows_unit_tests_with_dask.yml

```diff
@@ -0,0 +1,74 @@
+name: Windows Unit Tests (Dask Only)
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+  push:
+    branches:
+      - main
+
+jobs:
+  win_unit_tests:
```

Comment:
I'll file an issue for trying to reduce as many of the set-up steps as possible between the dask-only and non-dask tests. I think that'll pay off in the future.